### PR TITLE
Gapfill wrapper

### PR DESCRIPTION
**Pull request by msilvafe**
I'm adding a wrapper for the `tod_ops.gapfill` module that @bkelle and I have been using in our testing of the LAT + SAT preprocessing. I also added in a unit test script, and process to preprocess module. I tested the wrapper function on its own and running through preprocess. My own testing was following the same test in the unit test script which generates an axis manager with all of the detectors showing the same polynomial signal but with different glitches in each detector. Currently the polynomial gap filler is working very well for this test case but the pca model filling is failing so for now I have just commented out that test in the unit test script.
| Testing Function w/ Polynomial Filler  | Testing Function w/ PCA Filler  | Testing running through preprocess module |
| ------------- | ------------- | ------------- |
| gapfill_demo_poly | gapfill_demo_pca  | gapfill_demo_poly_preprocess |
| Glitch filling works well using polynomial filler. | PCA filling failing, @mhasself suggests poly filling first before running the PCA model fill but this returns a `TypeError` when run with the simple AxisManager that can be constructed as is done in the unit test script.  | Polynomial filler used in the preprocess module works well. |

All of the plots here and testing can be found in a notebook on simons1: `/home/msilvafe/notebooks/flp_dev/20230524_Test_Gapfill.ipynb` and the preprocess configs can also be found there.

### CODE REVIEW

**Comment by msilvafe** [comment]
@mhasself figured out that to avoid the poor gap filling from the pca module we first needed to 1) gapfill with the poly gap filler 2) run the pca 3) gap fill with the pca model. So I changed to that. Now both methods are working well:

| Testing Function w/ Polynomial Filler  | Testing Function w/ PCA Filler  |
| ------------- | ------------- |
| gapfill_demo_poly_post_fix | gapfill_demo_pca_post_fix |
| Glitch filling works well using polynomial filler. | Glitch filling works well using PCA filler. |

I was also getting a TypeError when running the pca gap filling because with all of the signals the same the pca would return complex eigenvalues, which couldn't be added to regular float64. So I updated the pca module to only return real eigenvalues per Matthew's suggestion. So everything should be working and ready for review post commit [f82eb6b](https://github.com/simonsobs/sotodlib/pull/476/commits/f82eb6b268732c795716196f6ba632a9af50cb8f).

File: sotodlib/tod_ops/gapfill.py
Comment (mhasself) [inline sotodlib/tod_ops/gapfill.py:3]: Use relative references for internal imports, `from . import pca`.

File: sotodlib/tod_ops/gapfill.py
Comment (mhasself) [inline sotodlib/tod_ops/gapfill.py:427]: Remove " is True"

File: sotodlib/tod_ops/gapfill.py
Comment (mhasself) [inline sotodlib/tod_ops/gapfill.py:442]: Not "else:"?

File: sotodlib/tod_ops/gapfill.py
Comment (mhasself) [inline sotodlib/tod_ops/gapfill.py:392]: "assuming glitch flags exist" -> "using pre-computed glitch flags provided by the caller."
"Top level function for filling glitches"  -> "Fill glitches in time-ordered data"

Mention that it uses polynomial filling, by default, but can use PCA optionally.

This function fills glitches, using either a polynomial or PCA-based approach.

File: sotodlib/tod_ops/gapfill.py
Comment (mhasself) [inline sotodlib/tod_ops/gapfill.py:413]: In other places we have used "wrap" to support the thing these two arguments are doing.  E.g.:

wrap=False -> no, don't merge it just return it
wrap=True -> yes, merge the new array and call it the default thing ("gap_filled" apparently)
wrap="my_other-name" -> yes, merge the new array and cal it "my_other_name"

Can the preprocess system cope with that setup?  Can you use generic "wrap" instead of specific "merge_filled, filled_name"?

File: sotodlib/tod_ops/gapfill.py
Comment (mhasself) [inline sotodlib/tod_ops/gapfill.py:445]: This bit, for polynomial filling, is the same code that is run in the PCA block.  So you could reduce code duplication and eliminate the `else` by doing:
```
    [process args]
    [polyfill]
    if use_pca:
       [pcafill]
    [wrap and return]
```

File: sotodlib/tod_ops/gapfill.py
Comment (mhasself) [inline sotodlib/tod_ops/gapfill.py:435]: It is annoying this needs to be converted to float32 ... will create an issue to resolve that.

**Review by mhasself — CHANGES_REQUESTED** [review]
Thanks for this -- will be very useful.

This PR seems to have mixed in resample.py as well.  I thought that was intended to be in a separate PR, so I've not reviewed it for now.

File: tests/test_tod_ops.py
Comment (mhasself) [inline tests/test_tod_ops.py:173]: This isn't really "in_place".  "in_place" would mean that tod.signal is modified directly, without creating a whole new vector for glitch_filled.  Also, there's no harm in return glitch_filled if it was also wrapped.

File: sotodlib/preprocess/processes.py
Comment (mhasself) [inline sotodlib/preprocess/processes.py:248]: Sorry, nitpicking now -- looks like the naming convention for preprocess modules is "ThingAction", so I would think this would be GlitchFill.

**Review by mhasself — APPROVED** [review]
A couple minor things (but approving pending those fixes).  Thanks for this, and for including tests.

File: tests/test_tod_ops.py
Comment (msilvafe) [reply tests/test_tod_ops.py:173]: Do you want me to make an in_place option or just change the comment to "test wrap new field"? Would you prefer I always return glitch_filled so instead of checking if the returned value is None here instead check if the new field exists in the axis manager?

Comment (mhasself) [reply tests/test_tod_ops.py:173]: Just change the comment.

Yes, I think it's better to always return glitch_filled, and instead check that the correct new field was created in the axisman.

**Pull request merged**
